File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/responses/user/UserResponse.java
package com.example.dalathasfarm.responses.user;

import com.example.dalathasfarm.models.Role;
import com.example.dalathasfarm.models.User;
import lombok.*;

import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserResponse {
    private Integer id;
    private String fullName;
    private String address;
    private String phoneNumber;
    private Date dateOfBirth;
    private String email;
    private Boolean isActive;
    private String profileImage;
    private Role roleName;

    public static UserResponse fromUser(User user) {
        return UserResponse
                .builder()
                .id(user.getId())
                .fullName(user.getFullName())
                .address(user.getAddress())
                .phoneNumber(user.getPhoneNumber())
                .dateOfBirth(user.getDateOfBirth())
                .email(user.getEmail())
                .profileImage(user.getProfileImage())
                .isActive(user.getIsActive())
                .roleName(user.getRole())
                .build();
    }
}

File: android/lib/responses/user_response.dart
import 'package:android/models/role.dart';

class UserResponse{
  final int id;
  final DateTime dateOfBirth;
  final String? address;
  final String phoneNumber;
  final bool isActive;
  final String fullName;
  final String? email;
  final Role role;

  UserResponse({
    required this.id,
    required this.fullName,
    this.email,
    required this.phoneNumber,
    this.address,
    required this.dateOfBirth,
    required this.role,
    required this.isActive,
  });

  factory UserResponse.fromJson(Map<String, dynamic> json) {
    return UserResponse(
      id: json['id'],
      fullName: json['fullName'],
      email: json['email'],
      phoneNumber: json['phoneNumber'],
      address: json['address'],
      dateOfBirth: DateTime.parse(json['dateOfBirth']),
      role: Role.fromJson(json['role']),
      isActive: json['isActive'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'fullName': fullName,
      'email': email,
      'phoneNumber': phoneNumber,
      'address': address,
      'dateOfBirth': dateOfBirth.toIso8601String(),
      'role': role.toJson(),
      'isActive': isActive,
    };
  }
}